protos/escrow.proto

```diff
@@ -0,0 +1,99 @@
+syntax = "proto3";
+import "google/protobuf/timestamp.proto";
+
+package btfs;
+
+service EscrowService{
+	rpc submitContracts(EscrowContractRequest) returns (EscrowContractResult) {}
+	rpc payIn(PayinRquest) returns (PayinResult) {}
+	rpc checkPayment(ContractID) returns (PayinStatus) {}
+	rpc cancelContracts(CancelContractRequest) returns (CancelContractResult) {}
+};
+
+enum TokenType {
+	BTT = 0;
+}
+
+enum Schedule {
+	MONTHLY = 0;
+	QUARTERLY = 1;
+	ANNUALLY = 2;
+}
+
+message ContractID {
+	bytes contract_id = 1;
+}
+
+message EscrowContract {
+	bytes contract_id = 1; // identity of a contract
+	bytes buyer_address = 2;
+	bytes buyer_signature = 3;
+	bytes seller_address = 4;
+	bytes seller_signature = 5; // the seller signature
+	bytes auth_address = 6; // authority who can cancel escrow
+	bytes auth_signature = 7;
+	int64 amount = 8;
+	int64 collateral_amount = 9;
+	int64 withhold_amount = 10; // escrow hold amount till the last layment
+	TokenType token_type = 11; // token type enum
+	Schedule payout_schedule = 12; // schedule enum
+	int32 num_payouts = 13; // numbers of payouts
+	bytes reference = 14; // reference like hash value etc
+}
+
+message EscrowContractRequest {
+	repeated EscrowContract contract = 1;
+}
+
+message Escrow {
+	bytes contract_id = 1;
+	int64 escrow_id = 2;
+	bytes from_address = 3;
+	bytes to_address = 4;
+	int64 amount = 5;
+	int64 collateral_amount = 6;
+}
+
+message EscrowContractResult {
+	bytes payin_id = 1; // for buyer to pay in a group of escrows
+	bytes escrow_address = 2;
+	repeated Escrow escrow = 3;
+	google.protobuf.Timestamp escrow_signed_time = 4;
+	bytes escrow_signature = 5;
+}
+
+message PayinRquest {
+	bytes payin_id = 1;
+	bytes buyer_address = 2;
+	bytes buyer_signature = 3;
+}
+
+message PayinResult {
+	bytes payin_id = 1;
+	int64 sequence_id = 2; // ledger sequence id
+	google.protobuf.Timestamp escrow_signed_time = 3;
+	bytes escrow_signature = 4;
+}
+
+message PayinStatus {
+	bool paid = 1;
+	int64 sequence_id = 2;
+}
+
+message CancelContractRequest {
+	repeated bytes contract_id = 1;
+	bytes auth_address = 2;
+	bytes auth_signature = 3; // authority signature
+}
+
+message CancelContractResult {
+	message Result {
+		bytes contract_id = 1;
+		bytes escrow_id = 2;
+		bool canceled = 3;
+	}
+	
+	Result result = 1;
+	google.protobuf.Timestamp escrow_signed_time = 2;
+	bytes escrow_signature = 3;
```

Comment:
Same question, there is no `escrow_address`, only `escrow_id`, maybe we only ever need `escrow_address`?

Reply:
fixed